src/sage/rings/polynomial/term_order.py

```diff
@@ -386,9 +386,15 @@
     'negwdeglex'    : 'Negative weighted degree lexicographic',
 }
 
+from sage.interfaces.singular import singular_version_number
+if int(singular_version_number()[0:3]) < 432 or (int(singular_version_number()[0:3]) == 432 and int(singular_version_number()[3:]) < 15):
+    invlex_singular_name = 'rp'
+else:
+    invlex_singular_name = 'ip'
+
 singular_name_mapping = {
     'lex'           : 'lp',
-    'invlex'        : 'rp',
+    'invlex'        : invlex_singular_name,
```

Comment:
Seems to be the only place we need a different code for different versions of singular.